### PR TITLE
Much work on the CLI

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs [hunk 1]
```diff
@@ -181,6 +182,12 @@ newtype PrivKeyFile
   = PrivKeyFile FilePath
   deriving (Eq, Show)
 
+newtype TxBodyFile = TxBodyFile FilePath
+  deriving (Eq, Show)
```
Comment (Jimbo4350) [inline cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:188]: @erikd should these two have `deriving newtype` ?

File: cardano-cli/test/cli/address-describe/run [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#!/bin/sh -u
```
Comment (Jimbo4350) [inline cardano-cli/test/cli/address-describe/run:1]: :100:

File: cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs (hunk 1 above)
Comment (erikd) [reply cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs:188]: Yeah probably, but not a big issue. Can be fixed in the next PR set.

File: cardano-api/test/Test/Cardano/Api/Gen.hs [hunk 3]
```diff
@@ -142,3 +133,55 @@ genTxUnsignedShelley =
...
+-- -------------------------------------------------------------------------------------------------
+
+genByteString :: Range Int -> Gen ByteString
+genByteString lenRange =
+  BS.pack <$> Gen.list lenRange (Gen.element [ '\x00' .. '\xff' ])
+
+-- Generates a fake TxId by applying the right hashing function to a random ByteString.
+genFakeTxId :: Gen TxId
+genFakeTxId =
+  TxId . {- Crypto. -} hashRaw identity <$> genByteString (Range.linear 10 50)
```
[the remark this answers is not in the document]
Comment (erikd) [reply cardano-api/test/Test/Cardano/Api/Gen.hs:168]: No that was intended as a helper for when `hashRaw` bubbles up from `base`.

File: cardano-api/src/Cardano/Api/Convert.hs [hunk 4]
```diff
@@ -0,0 +1,96 @@
+module Cardano.Api.Convert
+  ( addressFromHex
+  , addressToHex
+  , parseTxIn
+  , parseTxOut
+  , renderTxIn
+  , renderTxOut
```
[the remark this answers is not in the document]
Comment (erikd) [reply cardano-api/src/Cardano/Api/Convert.hs:7]: Agree, but will leave that for later. Tests will have to be moved as well.

**Pull request merged**